packages/contracts/contracts/VkRegistry.sol

```diff
@@ -10,7 +10,7 @@ import { DomainObjs } from "./utilities/DomainObjs.sol";
 /// @notice Stores verifying keys for the circuits.
 /// Each circuit has a signature which is its compile-time constants represented
 /// as a uint256.
-contract VkRegistry is Ownable(msg.sender), DomainObjs, SnarkCommon, IVkRegistry {
```

Comment:
After fixing the contract deployment and address fetching I came across another problem: the `VKRegistry` contract was setting `owner = safe-create-call-address`

This is because when the sponsored AA wants to deploy a contract sends a transaction to `safe-create-call-address` with the deploy info in the calldata. This makes the `safe-create-call-address` the owner of `VKRegistry`


To fix it I made the constructor receive a parameter that will set the owner. The deployment function takes the signer address and sends it as parameter in the constructor. Check the next comment for more details